### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by dorothykiz1**
@joyLubega
@nuwagalb @leni1 kindly review my work

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def post(self):
+        pass
+
+    def edit(self):
+        pass
+
+    def delete(self):
+        pass
+
```
Comment (leni1) [inline api/models.py]: @dorothykiz1 why are all the classes inheriting from `Question`? To me these would be separate classes in the case of `User` and `Question`; inheritance between `Answer` and `Question` is much more likely since there's overlap in their functions and maybe you'd only want to customize a few things in `Answer` that will not be needed in `Question`.

File: api/questions.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        return jsonify("Answered question"), 200
+    else:
+        return jsonify("Question index is out of range"), 504
+
+
+@app.route("/api/v1/questions/<int:id>", methods=["DELETE"])
+def delete_questions(id):
+    del questions[id]
+    return jsonify(questions), 200
+
```
Comment (leni1) [inline api/questions.py]: @dorothykiz1 as far as I can see, you're not using your classes here. The other thing is that your views should not have the data structure code and the corresponding operations; ideally this should be in your models (classes) and your views just pass along and return data.

File: tests/test_question.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            data=self.question)
+        self.assertEqual(result.status_code, 201)
+        res = self.client().delete('/api/v1/questions/1')
+        self.assertEqual(res.status_code, 200)
+        # Test to see if it exists, should return a 404
+        result = self.client().get('/api/v1/questions/1')
+    
+        self.assertEqual(result.status_code, 404)
+
+#execute tests
```
Comment (leni1) [inline tests/test_question.py:58]: @dorothykiz1 these tests look fine. Try to add tests for when fields are empty among other test cases.

**Comment by leni1** [comment]
@dorothykiz1 

1. Why are all the classes inheriting from `Question`? To me these would be separate classes in the case of `User` and `Question`; inheritance between `Answer` and `Question` is much more likely since there's overlap in their functions and maybe you'd only want to customize a few things in `Answer` that will not be needed in `Question`. 

2. Your views should not have the data structure code and the corresponding operations; ideally this should be in your models (classes) and your views just pass along and return data. 

3. There's no need to use `JSONResponse`; just make sure that the data you return is in dictionary format and you should be good to go. Again the code for this should be in your models not in the views themselves.

4. In future, when asking for reviews, do so when you have at least 5 or 6 commits in a pull request. Anything more than that, and it becomes difficult to review your code well since one can make comments on an outdated version of a file. 

Otherwise, keep up the good work. :clap: As a side note, try reviewing the Flask Documentation. In particular, the sample tutorial they have on their page for building a blog. Building an API is similar only that you'll be returning JSON responses rather than HTML files. It will help with your code organization. :smile:

**Comment by dorothykiz1** [comment]
Thanks @leni1 for taking your time to see all the shortcuts 
Let me get to work😊
